### ISSUE TITLE
Load state notifier before runApp? 

### ISSUE THREAD

**Issue by danieldoddjr** [rrousselGit/river_pod#1129]
I have a user state notifier that manages login and logout. If the user is logged in when the app starts I redirect to a logged in route. Because it needs to get the user from hive it is not ready in the time. How can I wait for the state notifier to do its thing before runApp starts the app? 

I have not found anything that describes the ability to load state-notifiers before starting the app.

**Comment by XuanTung95** [comment]
This question should be asked on stackoverflow.

**Comment by danieldoddjr** [comment]
If there is a process for doing this I think its worth documenting.

**Comment by XuanTung95** [comment]
Riverpod is designed to work with or without Flutter. You can check the document for the example of using it with dart only.

**Comment by raminsafaei** [comment]
There are multiple ways to do such a thing. If your state notifiers don't rely on riverpod(don't need ref, read etc.) you can use something like a valuenotifier to initialize app before running main MaterialApp. If you're looking to actually prevent flutter from drawing pixels until certain classes are loaded, you can use [Flutter native splash's latest version](https://github.com/jonbhanson/flutter_native_splash), which has a [neat way](https://github.com/jonbhanson/flutter_native_splash/blob/master/lib/flutter_native_splash.dart) to do this(or just copy the function and use it the way you like)

**Comment by rrousselGit** [comment]
Closing in favour of https://github.com/rrousselGit/river_pod/issues/57
